pallets/order-book/src/lib.rs

```diff
@@ -721,123 +730,80 @@ pub mod pallet {
 				.ok_or(Error::<T>::InvalidAssetId)?
 				.decimals;
 
-			convert_balance_decimals(from_decimals, to_decimals, ratio.ensure_mul_int(amount)?)
-				.map_err(DispatchError::from)
+			Ok(convert_balance_decimals(
+				from_decimals,
+				to_decimals,
+				ratio.ensure_mul_int(amount_from)?,
+			)?)
 		}
-	}
 
-	impl<T: Config> Pallet<T> {
-		fn is_valid_order(
-			currency_in: T::AssetCurrencyId,
-			currency_out: T::AssetCurrencyId,
-			buy_amount: T::Balance,
-			sell_rate_limit: T::SellRatio,
-			min_fulfillment_amount: T::Balance,
+		fn validate_amount(
+			amount_out: T::Balance,
+			min_fulfillment_amount_out: T::Balance,
 			min_order_amount: T::Balance,
 		) -> DispatchResult {
-			ensure!(currency_in != currency_out, Error::<T>::ConflictingAssetIds);
-
-			ensure!(
-				buy_amount != <T::Balance>::zero(),
-				Error::<T>::InvalidBuyAmount
-			);
-
-			ensure!(
-				min_fulfillment_amount != <T::Balance>::zero(),
-				Error::<T>::InvalidMinimumFulfillment
-			);
 			ensure!(
-				sell_rate_limit != T::SellRatio::zero(),
-				Error::<T>::InvalidMaxPrice
-			);
-
-			ensure!(
-				buy_amount >= min_fulfillment_amount,
+				amount_out >= min_fulfillment_amount_out,
 				Error::<T>::InvalidBuyAmount
 			);
 
 			ensure!(
-				buy_amount >= min_order_amount,
+				amount_out >= min_order_amount,
 				Error::<T>::InsufficientOrderSize
 			);
 
 			Ok(())
 		}
 
 		fn inner_update_order(
-			account: T::AccountId,
-			order_id: T::OrderIdNonce,
-			buy_amount: T::Balance,
-			sell_rate_limit: T::SellRatio,
-			min_fulfillment_amount: T::Balance,
-			validate: impl FnOnce(&OrderOf<T>) -> DispatchResult,
+			mut order: OrderOf<T>,
+			amount_out: T::Balance,
+			price: OrderPrice<T::SellRatio>,
+			min_amount_out: T::Balance,
 		) -> DispatchResult {
-			let max_sell_amount = <Orders<T>>::try_mutate_exists(
-				order_id,
-				|maybe_order| -> Result<T::Balance, DispatchError> {
-					let order = maybe_order.as_mut().ok_or(Error::<T>::OrderNotFound)?;
+			let min_fulfillment_amount_out = T::DecimalConverter::to_asset_balance(
+				T::MinFulfillmentAmountNative::get(),
+				order.asset_out_id,
+			)?;
```

Comment:
NOTE: As raised on Slack, this solely applies redenomination but does not determine whether the outgoing amount is worth 10 CFG. I suppose I did not bother with this because CFG internally is treated as 1 USD and we assume the OrderBook to solely deal with stable currencies. Not sure if we want to make it more explicit.

Reply:
Yes, this is a legacy issue. In order to make a currency conversion, we would have to require conversion rates for `CFG <-> asset_out_id` which increases the overhead. Maybe adding a comment `// Implicitly assuming 1 CFG = 1 asset_out for min fulfilment amount` suffices here because implicitly treating CFG as 1 unit of stable currency is fine for now for the purpose of determining the minimum fulfilment amount?!

WDYT @mustermeiszer?

Reply:
So, we need to choose between:
- Used 1 unit of `currency_out` as the min amount (current behavior, I think)
- Used a conversion to require an amount of `currency_out` equivalent to 1 CFG.